### PR TITLE
feat(new): Added Azure.ServiceBus.GeoReplica

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Working towards #2988

Added Azure.ServiceBus.GeoReplica.

## PR Checklist

- [x] PR has a meaningful title
- [ ] Summarized changes
- [ ] Change is not breaking
- [ ] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [ ] Unit tests created/ updated
  - [ ] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by BenjaminEngeset** [comment]
Hi @BernieWhite. I need some feedback from you on how to construct the rule body on this in the best way possible, also given the updates that will come so we don't have to do major refactor. The documentation states specifically `Currently only a single secondary is supported.`

Do you have any thoughts on how the rule body can look like? I have uploaded the draft design in this PR, but I feel it is to static and if the order is wrong in the array it will not work correctly.

https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-geo-replication#using-bicep-template

**Comment by BernieWhite** [comment]
> Hi @BernieWhite. I need some feedback from you on how to construct the rule body on this in the best way possible, also given the updates that will come so we don't have to do major refactor. The documentation states specifically `Currently only a single secondary is supported.`
> 
> Do you have any thoughts on how the rule body can look like? I have uploaded the draft design in this PR, but I feel it is to static and if the order is wrong in the array it will not work correctly. Maybe two contains methods can be used? If the operand is an array of strings, only one string must contain the specified string.
> 
> https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-geo-replication#using-bicep-template

Thanks @BenjaminEngeset I think the can primarily count how many entries are in `properties.geoDataReplication.locations`. i.e should be 2 or greater. The specific locations or roles and their order don't really matter and could easily change.

**Comment by BenjaminEngeset** [comment]
Ready for review @BernieWhite.

I noticed one thing that made me unsure about using YAML and that was that once the `AllOf` operator finds a fail, it will return and yield only that. So when the customer thinks that it is only the SKU that is the problem, next time it will yield the geo-replication configuration. I want both failures to be yielded. What do you think, is there anything we can do or should we move over to PowerShell?

**Comment by BernieWhite** [comment]
> Ready for review @BernieWhite.
> 
> I noticed one thing that made me unsure about using YAML and that was that once the `AllOf` operator finds a fail, it will return and yield only that. So when the customer thinks that it is only the SKU that is the problem, next time it will yield the geo-replication configuration. I want both failures to be yielded. What do you think, is there anything we can do or should we move over to PowerShell?

That's fair. We should just improve that behaviour in PSRule.

File: docs/en/rules/Azure.ServiceBus.GeoReplica.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+Consider replicating service bus namespaces to improve resiliency to region outages.
+
+## EXAMPLES
+
+### Configure with Azure template
+
+To deploy namespaces that pass this rule:
+
+- Set `sku.name` to `Premium`.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.ServiceBus.GeoReplica.md:47]: It's not possible to easily set the SKU to `Premium` if it's not already premium. There is a migration required to a new Premium namespace from an existing Standard tier.

```suggestion
- Set the `sku.name` property to `Premium`. This only applies to new namespaces.
  - For existing namespaces first, migrate to a namespace deployed with a premium SKU.
```

File: docs/en/rules/Azure.ServiceBus.GeoReplica.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    }
+  }
+}
+```
+
+### Configure with Bicep
+
+To deploy namespaces that pass this rule:
+
+- Set `sku.name` to `Premium`.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.ServiceBus.GeoReplica.md:85]: It's not possible to easily set the SKU to `Premium` if it's not already premium. There is a migration required to a new Premium namespace from an existing Standard tier.

```suggestion
- Set the `sku.name` property to `Premium`. This only applies to new namespaces.
  - For existing namespaces first, migrate to a namespace deployed with a premium SKU.
```

File: docs/en/rules/Azure.ServiceBus.GeoReplica.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+This feature is only supported for the `Premium` SKU.
+
+This feature is currently in preview.
+
+## LINKS
+
+- [RE:05 Redundancy](https://learn.microsoft.com/azure/well-architected/reliability/redundancy)
+- [Geo-replication](https://learn.microsoft.com/azure/service-bus-messaging/service-bus-geo-replication)
+- [Configure geo-replication](https://learn.microsoft.com/azure/service-bus-messaging/service-bus-geo-replication#setup)
```
Comment (BernieWhite) [inline docs/en/rules/Azure.ServiceBus.GeoReplica.md:127]:
```suggestion
- [Configure geo-replication](https://learn.microsoft.com/azure/service-bus-messaging/service-bus-geo-replication#setup)
- [Migrate existing Azure Service Bus standard namespaces to the premium tier](https://learn.microsoft.com/azure/service-bus-messaging/service-bus-migrate-standard-premium)
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Thanks @BenjaminEngeset. Great work, only a few minor suggestions.

**Comment by BenjaminEngeset** [comment]
Thank you for the feedback. The requested changes have been implemented, @BernieWhite.

**Review by BernieWhite — APPROVED** [review]
Thanks @BenjaminEngeset. All good to merge.

**Pull request merged**